### ISSUE TITLE
asinh of smallish value is infinite

### ISSUE THREAD

**Issue by EbTech** [rust-lang/rust#73642]
With `rustc` version 1.44.1, the following expression evaluates to minus infinity, instead of the correct value of approximately -18.7:

```rust
fn main() {
    let x = (-67886600.94408971f64).asinh();
    assert!(!x.is_infinite());
}
```
Perhaps taking the absolute value of the first `self` in the [implementation of `asinh`](https://doc.rust-lang.org/src/std/f64.rs.html#836) would make it more numerically stable?

**Comment by robojumper** [comment]
This is fixed on nightly with #72486. ([playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=f462f55e745e98f4e10ed7f03eb0ce96))

**Comment by EbTech** [comment]
Ah sorry, I should have checked!
